### PR TITLE
Cleanup cloud/services

### LINKED ISSUE #1001: Clean up ./cloud/services
/kind cleanup

**Describe the solution you'd like**
A while back, the code used to interact with Azure under `./cloud/services` was interfaced to allow some separation of concern between the controller / scopes and the types from the Azure SDK. With the advent of the `azure.Service` interface, Azure SDK / client code doesn't leak into the controllers. In fact, there should not longer be dependencies on the code that directly interacts with Azure. We should use this as an opportunity to crystallize this design goal of refactoring to `azure.Service` by cleaning up the service internals.

Current structure:
```
cloud/services/scalesets/
├── client.go
├── mock_scalesets
│   ├── client_mock.go
│   ├── doc.go
│   └── scalesets_mock.go
├── service.go
├── vmss.go
└── vmss_test.go
```

Proposed structure:
```
cloud/services/scalesets/
├── client.go  <-- don't export this; make it private
├── mock_scalesets
│   ├── client_mock.go
│   ├── doc.go
│   └── scaleset_mock.go
├── scaleset.go <-- includes service and scope
└── scaleset_test.go
```
The only thing exported from the new structure is the `Service` struct and the `Scope` interface for each service. This will ensure that the encapsulation will not be violated and guide future contributors.

### PR DESCRIPTION
**Pull request by shysank**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Cleanup cloud/services

Following #1013 , this pr cleans up rest of the cloud services (except `agentpools`, `managedclusters` since it's still using `azure.OldService`). The following are the summary of changes in each service:
- Un-export `Client` interface. There are a few exceptions here namely: `loadbalancers`, `networkinterfaces`, `publicips`, `scalesets`, `subnets`, `virtualmachines`, `virtualnetworks`. This is because these clients are also used in other services.
- Un-export other service's clients that are consumed
- Rename `NewService` to `New`
- Move `Scope` interface and `Service` struct definitions from `service.go` to `$service_name$.go`
- Fix import ordering

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1001 

**Special notes for your reviewer**:
 Every service cleanup is in a separate commit for now for easier review. Happy to split it into separate prs if needed.

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Cleanup cloud/services
```

### CODE REVIEW

**Review by CecileRobertMichon — COMMENTED** [review]
/ok-to-test

**Comment by CecileRobertMichon** [comment]
@shysank appologies, looks like #1022 caused a few conflicts

**Comment by shysank** [comment]
@CecileRobertMichon Thanks for letting me know, I've rebased with master. Although I think it's going to be more painful when [this](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1025/) is merged :(

**Comment by alexeldeib** [comment]
> Although I think it's going to be more painful when this is merged :(

I took a quick glance and while there would be lots of conflicts I don't think the diff is too bad and the changes don't affect each other too much. Happy to rebase when this is in since it's big but mostly cleanup.

File: cloud/services/bastionhosts/mocks_bastionhosts/client_mock.go [hunk 1]
```diff
@@ -27,31 +27,31 @@ import (
 	reflect "reflect"
 )
 
-// MockClient is a mock of Client interface.
-type MockClient struct {
+// Mockclient is a mock of client interface.
+type Mockclient struct {
```
Comment (CecileRobertMichon) [inline cloud/services/bastionhosts/mocks_bastionhosts/client_mock.go:31]: this looks a little strange, I would have expected camel case... but I know it's auto-generated so nothing we can do

**Review by CecileRobertMichon — APPROVED** [review]
/lgtm

**Pull request merged**